### PR TITLE
stdenv/check-meta: change to allowlist and blocklist

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
It felt significant to make this rename...
I haven't thought of the way to make it throw with instructions to change to the new attr, and I still need to make documentation for it. For now it will be a draft until I get back to it. But if anyone wants to finish that go ahead.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/stdenv/generic/check-meta.nix [hunk 1]
```diff
@@ -16,8 +16,8 @@ let
   allowUnfree = config.allowUnfree or false
     || builtins.getEnv "NIXPKGS_ALLOW_UNFREE" == "1";
 
-  whitelist = config.whitelistedLicenses or [];
-  blacklist = config.blacklistedLicenses or [];
+  allowlist = config.allowlistedLicenses or [];
+  blocklist = config.blocklistedLicenses or [];
```
Comment (grahamc) [inline pkgs/stdenv/generic/check-meta.nix:20]: What about providing backwards compatibility via:

```suggestion
  allowlist = config.allowlistedLicenses or config.whitelistedLicenses or [];
  blocklist = config.blocklistedLicenses or config.blacklistedLicenses or [];
```

**Review by grahamc — APPROVED** [review]
This looks good. One way we could merge it right away is by falling back to the old names if the new names are not present. It would not throw / warn, but it would get us a step further.

File: pkgs/stdenv/generic/check-meta.nix (hunk 1 above)
Comment (worldofpeace) [reply pkgs/stdenv/generic/check-meta.nix:20]: I had this but deleted it, but if u think it's fine let's do it :smile:

**Comment by worldofpeace** [comment]
I can make a release note later. Will also do nixos option renames.

**Comment by mohe2015** [comment]
Too late but why not `allowedLicences` and `forbiddenLicenses`?

**Pull request merged**